### ISSUE TITLE
Consistent Indentation in Multi-line Conditionals

### ISSUE THREAD

**Issue by Zamiell** [eslint/eslint#13153]
**Please describe what the rule should do:**

The rule should enforce consistent indentation in a multi-line conditional.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[X] Enforces code style (layout)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

As you probably know, it is common to break up long conditionals into multiple lines.
For example, in the Airbnb style guide, they prescribe breaking up a long conditional like so:

```js
// Bad code, this breaks the "max-len" eslint rule
if (longCondition1 && longCondition2 && longCondition3 && longCondition4 && longCondition5) {
  console.log('horray, all the conditions are true!')
}
```
-->
```js
// Good code
if (
  longCondition1 // Also note that we use a 2 space indent for each condition
  && longCondition2 // This is consistent with the Airbnb "indent" eslint rule
  && longCondition3
  && longCondition4
  && longCondition5
) {
  console.log('horray, all the conditions are true!')
}
```

(This is documented in [issue 1380](https://github.com/airbnb/javascript/issues/1380).)

However, what about code like this:

```s
// Bad code
if (
  longCondition1
  && longCondition2
   && longCondition3 // Oops, I accidentally added an extra space here at the beginning of the line
  && longCondition4 // Now I no longer have consistent indentation
  && longCondition5 // This code passes the "indent" rule somehow?
) {
  console.log('horray, all the conditions are true!')
}
```

1) I'm guessing that the "indent" rule doesn't care about multi-line conditionals, in order to allow programmers to write code in the Python style, like:

```js
if (longCondition1
    longCondition2) {
```

So, I'm guessing it is working as intended. Fine.

2) But when looking at my "Bad code" snippet above, the extra space is **clearly** a bug in the code. I was not able to find *any* eslint rules to enforce consistent indentation in conditionals, which was honestly surprising to me, as it seems like an obvious thing to want to enforce.

Another example, with a hypothetical "consistent-indent-multiline-conditionals" set to a value of 2:

```js
// Bad code
if (
  longCondition1
  && longCondition2
  && (
  longCondition3 // Needs 4 spaces instead of 2
  || longCondition4
  )
) {
  foo();
}
```

```js
// Good code
if (
  longCondition1
  && longCondition2
  && (
    longCondition3
    || longCondition4
  )
) {
  foo();
}
```

**Why should this rule be included in ESLint (instead of a plugin)?**

This seems like trivial / basic functionality that everyone who happens to use the Airbnb-style multi-line conditional statements would want. And Airbnb happens to be the the most popular JavaScript style guide in the world, as far as I can tell.

**Is there a real-world case where this rule would be useful?**

Yes! When using the "eslint --fix" feature to transform my codebase from 4 space indent to 2, it generated a crapton of bugs - there was incorrect spacing on **every single multi-line conditional** throughout my entire project. I can understand that eslint might not be smart enough to figure out how to automatically indent multi-line conditionals (that's a separate issue), but at the very least there should be some kind of eslint rule to alert me that things are messed up!

**Are you willing to submit a pull request to implement this rule?**

No, my experience with JavaScript is quite limited, so I'll hopefully be able to recruit someone else to do it.

**Comment by mdjermanovic** [comment]
Hi @Zamiell, thanks for the issue!

This might be a duplicate of #12248 (see [this comment](https://github.com/eslint/eslint/issues/12248#issuecomment-542487158)).

**Comment by Zamiell** [comment]
Indeed, it sounds like #12248 is just a more-generic description of the functionality that I want, including assignment expressions in addition to multi-line conditions.

It appears as if the rule was championed on Nov 1, 2019 but no-one has submitted any code or commented since then. Is there someone I can donate to specifically on the ESLint team to get them to work on this?

**Comment by mdjermanovic** [comment]
#12248 will be implemented if and when it gets accepted.

Since it is an enhancement (rule change), per our [evaluation process](https://eslint.org/docs/maintainer-guide/issues#accepting-issues) the issue also needs 3 :+1: from other team members.

**Comment by Zamiell** [comment]
@mdjermanovic Thanks for the reply; I'll close this issue now then. On issue #12248, I pinged kaicataldo, but he has not responded for 9 days. Is there anything I can do to kick-start the process of getting 3 ESLint team members to approve it?
